### PR TITLE
Fix #296: Introspection on default value for input object is not following GraphQL spec

### LINKED ISSUE #296: Introspection on default value for input object is not following GraphQL spec
I defined an input object with a `HashMap` as default value. Introspection on the default value gives the `toString()` value of the `HashMap` (see `graphql.introspection#__InputValue`). However, [GraphQL specification](http://facebook.github.io/graphql/#sec-The-__InputValue-Type) requires the string to be encoded using GraphQL language.

As a result, GraphiQL would fail because of the incorrect introspection result.

### PR DESCRIPTION
**Pull request by dminkovsky**
Relevant lines: https://github.com/graphql-java/graphql-java/blob/6b94b458f6738bde331444d307d1081aecb6ab10/src/main/java/graphql/introspection/Introspection.java#L84-L89

I think this will require https://github.com/graphql-java/graphql-java/issues/126

### CODE REVIEW

File: src/test/groovy/graphql/Issue296.groovy [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+import static graphql.Scalars.GraphQLString
+import static graphql.schema.GraphQLArgument.newArgument
+import static graphql.schema.GraphQLFieldDefinition.newFieldDefinition
+import static graphql.schema.GraphQLInputObjectField.newInputObjectField
+import static graphql.schema.GraphQLInputObjectType.newInputObject
+import static graphql.schema.GraphQLObjectType.newObject
+import static graphql.schema.GraphQLSchema.newSchema
+
+class Issue296 extends Specification {
```
Comment (dminkovsky) [inline src/test/groovy/graphql/Issue296.groovy]: Use `TestUtil.schemaWithInputType(inputType)`

File: src/test/groovy/graphql/Issue296.groovy [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+    def "test"() {
+
+        def graphql = GraphQL.newGraphQL(newSchema()
```
Comment (dminkovsky) [inline src/test/groovy/graphql/Issue296.groovy]: Use `TestUtil.schemaWithInputType(inputType)`

**Comment by rayngwf** [comment]
This unit test checks the exact reported issue. Thanks.

To make unit test for default value more complete, we should include testing with other value types as well, including list and enum etc.

**Comment by dminkovsky** [comment]
Yes definitely.

чт, 12 янв. 2017 г. в 20:50, rayngwf <notifications@github.com>:

**Comment by bbakerman** [comment]
This PR fixes this issue and adds a shed load more functionality at the same time

https://github.com/graphql-java/graphql-java/pull/388/files

**Comment by bbakerman** [comment]
This is no longer needed as is

**Pull request closed without merging**